packages/agent/src/agent/http/index.ts

```diff
@@ -1398,7 +1398,6 @@ export function calculateIngressExpiry(
   maxIngressExpiryInMinutes: number,
   timeDiffMsecs: number,
 ): Expiry {
-  return Expiry.fromDeltaInMilliseconds(
-    maxIngressExpiryInMinutes * MINUTE_TO_MSECS + timeDiffMsecs,
-  );
+  const roundedExpiry = Expiry.fromDeltaInMilliseconds(maxIngressExpiryInMinutes * MINUTE_TO_MSECS);
+  return roundedExpiry.addMilliseconds(timeDiffMsecs);
```

Comment:
I'd still round at the very end for the sake of caching, but make sure the rounding strategy (round to seconds vs round to minutes) is based on `maxIngressExpiryInMinutes` only, i.e., not affected by `timeDiffMsecs`.